main.go

```diff
@@ -59,6 +59,11 @@ func cmd() *cobra.Command {
 			if err != nil {
 				return err
 			}
+			// Require `upstream-provider-name` to be set
+			if context.UpstreamProviderName == "" {
+				return errors.New("`upstream-provider-name` must be provided")
+			}
```

Comment:
If possible, can we do this within our CLI framework (cobra)?

If we are declaring this command required, should we make it positional?

Reply:
Good question, I'm not sure. I think it's better to specify it in the config file (which sets the corresponding flag), but if not, require the flag? Unless there's a benefit to having it positional

Reply:
We can leave it a flag. Positional just helps with brevity.

Reply:
Regardless, there is a better way to make this required: https://pkg.go.dev/github.com/spf13/cobra#MarkFlagRequired

Reply:
I tried `MarkFlagRequired` - it doesn't seem to work when you set `PersistentPreRunE` (https://stackoverflow.com/questions/38105859/make-a-cobra-command-flag-required)

Reply:
It works fine. Validation runs after `PersistentPreRunE`. If that is how `cobra` thinks about validation, we should probably move such validation into our `Run` func. Sorry for the back and forth.